src/android/SplashScreen.java

```diff
@@ -151,13 +193,40 @@ public Object onMessage(String id, Object data) {
         }
         return null;
     }
+    
+    @Override
+    public void onConfigurationChanged(Configuration newConfig) {
+        super.onConfigurationChanged(newConfig);
+
+        // Reload splash drawable when orientation changes if so configured.
+        if (newConfig.orientation != orientation) {
+            orientation = newConfig.orientation;
+            reloadDrawable();
+        }
+    }
 
-    private void removeSplashScreen() {
+    private void reloadDrawable() {
+        if (isReloadOnOrientationChange() && splashImageView != null) {
```

Comment:
ImageView.setImageDrawable documentation does not say anything about how smart it is about reloading and repainting the drawable, so I assumed the worst and added an explicit preference. I originally tried using ImageView.setImageResource, but that does nothing if resource ID did not change. To fix that I used ImageView.setImageDrawable and it reloaded the drawable, but I can only assume that it reloads it every time.